tests/storage.spec.ts

```diff
@@ -0,0 +1,204 @@
+/// <reference path="mochainon.d.ts" />
+
+import * as FsModule from 'fs';
+import * as m from 'mochainon';
+import * as path from 'path';
+import * as ResinSettingsClientModule from 'resin-settings-client';
+
+import getLocalStorage = require('../lib/local-storage');
+import getStorage = require('../lib/storage');
+
+const IS_BROWSER = typeof window !== 'undefined';
+
+let dataDirectory: string | undefined;
+let fs: typeof FsModule;
+if (!IS_BROWSER) {
+	// tslint:disable no-var-requires
+	fs = require('fs');
+	const settings: typeof ResinSettingsClientModule = require('resin-settings-client');
+	dataDirectory = settings.get<string>('dataDirectory');
+}
+
+const localStorage = getLocalStorage(dataDirectory);
+const storage = getStorage({ dataDirectory });
+
+// tslint:disable no-unused-expression
+
+describe('Storage:', () => {
+	describe('given numbers', () => {
+		it('should be able to save a float number', () =>
+			storage.set('pi', 3.14).then(() => {
+				return m.chai
+					.expect(storage.get('pi'))
+					.to.eventually.equal(3.14)
+					.then(() => storage.remove('pi'));
```

Comment:
I know it's preexisting, but it might simplify all this nicely to drop the `.remove` in every test and instead have a `beforeEach` that calls `storage.clear()`. Right now, any failing assertion won't run the subsequent remove, and can break the next test (potentially breaking a whole chain of them, leaving you in a proper mess)